src/draw/lv_image_decoder.c

```diff
@@ -233,6 +249,34 @@ void lv_image_decoder_set_close_cb(lv_image_decoder_t * decoder, lv_image_decode
     decoder->close_cb = close_cb;
 }
 
+void lv_image_decoder_set_cache_free_cb(lv_image_decoder_t * decoder, lv_cache_free_cb_t cache_free_cb)
+{
+    decoder->cache_free_cb = cache_free_cb;
+}
+
+lv_cache_entry_t * lv_image_decoder_add_to_cache(lv_image_decoder_t * decoder,
+                                                 lv_image_decoder_cache_data_t * search_key,
+                                                 const lv_draw_buf_t * decoded, void * user_data)
+{
+    lv_cache_entry_t * cache_entry = lv_cache_add(img_cache_p, search_key, NULL);
+    if(cache_entry == NULL) {
+        return NULL;
+    }
+
+    lv_image_decoder_cache_data_t * cached_data;
```

Comment:
It's an image cache, let's call it `lv_image_cache_data_t`

Reply:
great idea!